### DISCUSSION TITLE
Fear of Overprovisioning

### DISCUSSION THREAD

**Discussion by guettli** [topolvm/topolvm#918] [Q&A] [answered]
I have fear of using overprovisioning. We plan to use snapshots, and this needs thin-pools. I read the docs, but still I am unsure.

In the [thin-volumes](https://github.com/topolvm/topolvm/blob/main/docs/proposals/thin-volumes.md#capacity-awareness) proposal I read:

> 
> Capacity awareness
> 
> Topolvm uses capacity awareness to ensure that new pod is created on the node where available capacity of the volume group is larger than the requested capacity. capacity.topolvm.cybozu.com/<device-class> annotation is added on each node which consists of the available capacity on the volume group.
> Creation of thin logical volumes doesn't update the available device class capacity until the user actually uses storage on the thin volumes.
> So we need to ensure that all the thin logical volumes don't get created on the same node.
> This can be done by using the overprovisionRatio
> For device classes having thin pools, the value of capacity.topolvm.cybozu.com/<device-class> annotation will be calculated as: (thin-pool-size * overprovisionedRatio)-(sum-of-all-thin-lv-virtual-sizes-in-thin-pool)
> Note: Available thin pool on the volume groups will be obtained from lvmd.yaml config file.
> 

Up to now we plan to use the default capacity scheduler of Kubernetes, no the topolvm scheduler.

Imagine I have a 100 GB disk, and one VG and one thin-pool which use these 100 GB.

One PV uses 90 GB, and overprovisionedRatio is 1.0

Can a snapshot be created in this case?

If "Yes", then (thin-pool-size * overprovisionedRatio)-(sum-of-all-thin-lv-virtual-sizes-in-thin-pool) would be negative.

Is there a way to see the actual free disk space in an annotation, or is that only visible in the prometheus metrics?

**Comment by toshipp** [comment, marked as answer]
> Can a snapshot be created in this case?

I'm not confident though, kube-scheduler is unrelated to snapshots, Yes.

> Is there a way to see the actual free disk space in an annotation, or is that only visible in the prometheus metrics?

The capacity annotation does not reflect actual space for thinpool and is not expected to be used by end users.
You can calculate actual free space using `topolvm_thinpool_size_bytes` and `topolvm_thinpool_data_percent`, IMO.
